contracts/jobs/JobsManager.sol

```diff
@@ -70,94 +163,300 @@ contract JobsManager is IJobsManager, Verifiable, Controllable {
      */
     function job(string _streamId, string _transcodingOptions, uint256 _maxPricePerSegment) external returns (bool) {
         address electedTranscoder = bondingManager().electActiveTranscoder(_maxPricePerSegment);
+        /* There must be an elected transcoder */
+        require(electedTranscoder != address(0));
+
+        Job storage job = jobs[numJobs];
+        job.jobId = numJobs;
+        job.streamId = _streamId;
+        job.transcodingOptions = _transcodingOptions;
+        job.maxPricePerSegment = _maxPricePerSegment;
+        job.broadcasterAddress = msg.sender;
+        job.transcoderAddress = electedTranscoder;
+
+        NewJob(electedTranscoder, msg.sender, numJobs);
+
+        // Increment number of created jobs
+        numJobs = numJobs.add(1);
+
+        return true;
+    }
+
+    /*
+     * @dev End a job. Can be called by either a broadcaster or transcoder of a job
+     * @param _jobId Job identifier
+     */
+    function endJob(uint256 _jobId) external returns (bool) {
+        Job storage job = jobs[_jobId];
+
+        // Job must not already have an end block
+        require(job.endBlock == 0);
+        // Sender must be the job's broadcaster or elected transcoder
+        require(job.broadcasterAddress == msg.sender || jobs[_jobId].transcoderAddress == msg.sender);
```

Comment:
`jobs[_jobId].transcoderAddress == msg.sender` can be simplified to `job.transcoderAddress == msg.sender`